### PR TITLE
requests

### PR DESCRIPTION
**Pull request by FuhuXia**
Related to
- https://github.com/GSA/data.gov/issues/4053

Not sure why `urllib` is not able to get through proxy, but `requests` is ok. 

If we take this approach, there are some other places where `urllib` need to be changed to `requests`.

### CODE REVIEW

File: ckanext/datajson/blueprint.py [hunk 1]
```diff
@@ -333,18 +333,25 @@ def validator():
...
 
-            import urllib.request
-            import urllib.parse
-            import urllib.error
-            import json
+            import requests
+            try:
+                from requests.exceptions import JSONDecodeError
+            except ImportError:
+                from simplejson.scanner import JSONDecodeError
```
[the remark this answers is not in the document]
Comment (FuhuXia) [reply ckanext/datajson/blueprint.py:340]: It is based on trial and error. In a simple venv `requests.exceptions.JSONDecodeError` is raised, inside ckan docker env `simplejson.scanner.JSONDecodeError` is raised. This try...except get both working.

File: ckanext/datajson/harvester_datajson.py [hunk 2]
```diff
@@ -26,34 +27,29 @@ def info(self):
         }
 
     def load_remote_catalog(self, harvest_job):
-        req = urllib.request.Request(harvest_job.source.url)
+        url = harvest_job.source.url
         # todo: into config and across harvester
-        req.add_header('User-agent', 'Data.gov/2.0')
+        headers = {'User-Agent': 'Data.gov/2.0'}
```
[the remark this answers is not in the document]
Comment (FuhuXia) [reply ckanext/datajson/harvester_datajson.py:32]: http headers are case-insensitive, so either case works, but `User-Agent` is more correct than `User-agent`.

File: ckanext/datajson/harvester_datajson.py [hunk 3]
```diff
@@ -26,34 +27,29 @@ def info(self):
...
+            response.raise_for_status()
+        except requests.exceptions.HTTPError as e:
+            self._save_gather_error("HTTPError getting json source: %s." % (e), harvest_job)
+            return []
+
+        try:
+            datasets = response.json()
+        except JSONDecodeError as e:
+            self._save_gather_error("JSONDecodeError loading json. %s" % (e), harvest_job)
+            return []
```
[the remark this answers is not in the document]
Comment (FuhuXia) [reply ckanext/datajson/harvester_datajson.py:52]: One line in one try is more explicit so we know which line is causing which error. Multiple lines in one try block is the lazy way of doing things, it is ok if we don't care about knowing exact error source, especially when we have a catch all exception the end.

**Review by nickumia-reisys — APPROVED** [review]
🍔

**Pull request merged**